src/components/structures/MatrixChat.js

```diff
@@ -126,6 +143,33 @@ module.exports = React.createClass({
         }
     },
 
+    _registerAsGuest: function() {
+        var self = this;
+        var config = this.props.config;
+        console.log("Doing guest login on %s", config.default_hs_url);
+        MatrixClientPeg.replaceUsingUrls(
+            config.default_hs_url, config.default_is_url
+        );
+        MatrixClientPeg.get().registerGuest().done(function(creds) {
```

Comment:
Currently synapse does not require user interactive auth for guest registration. I don't want to add support for that when we have no real way of testing it, especially given it functions in its current form. Another PR can add support for this if and when we decide to go down that road.